rfcs/text/0000-rfc-application-layer-bundle-types-and-definitions.md

```diff
@@ -0,0 +1,431 @@
+### DCP PR:
+
+Link to document view: [here](https://github.com/HumanCellAtlas/dcp-community/blob/d2da58b16efd80fa3e31ed2194169efd79b02aa2/rfcs/text/0000-rfc-application-layer-bundle-types-and-definitions.md)
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# Application Layer Bundle Types and Definitions
+
+
+## Summary
+
+This RFC is offered as a refinement and generization of the [RFC: HCA DCP Bundle Types and Definitions]( https://github.com/HumanCellAtlas/dcp-community/pull/93). 
+
+This RFC describes:
+
+1. An application/metadata level mechanism to define bundle types and a simple bundle type hierarchy.
+
+1. An application/metadata level method to specify the type of a bundle.
+ 
+1. Several bundle types designed to match current usage and near term expected usage. 
+
+### Key Differences from RFC: HCA DCP Bundle Types and Definitions
+
+This proposal differs from [RFC: HCA DCP Bundle Types and Definitions](https://github.com/HumanCellAtlas/dcp-community/pull/93/files) mainly in that:
+
+1. Type information is added to a type.json  metadata file rather than added to the DSS bundle.json file.
+
+1. Type information is expressed in JSON rather than of [RFC 7231](https://tools.ietf.org/html/rfc7231#section-3.1.1.1) media type syntax.
+
+1. The schema and bundle types are represented as JSON schema in the metadata schema repo and documented on the Data Portal rather than maintained in a DSS registry.
+
+1. Additional background and motivation is presented.
+
+
+## Author(s)
+
+ [Dave Rogers](mailto:dave@clevercanary.com)
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+When dealing with schemaless data stores such as DSS, the responsibility for establishing the data schema and ensuring data validity becomes the responsibility of client applications writing data into the data store.
+
+If the application does not specify a way to define and implement types, it is possible for multiple “ad hoc” types or document shapes to evolve over time that are not well documented and are difficult to describe, validate, and query. 
+
+To address this, this RFC:
+
+
+1. Proposes a method of defining application level data types for DSS bundles.
+
+
+1. Proposes a method of implementing application level data types for DSS bundles.
+
+
+1. Proposes a simple type hierarchy for bundles that is closely tied to the application domain. 
+
+As the proposed updates to the bundle schema are at the application/metadata layer rather than the DSS/storage layer, no changes to DSS are envisioned, except for perhaps defining a new HCA file type and adding its contents to its index
```

Comment:
"new HCA file type". Do you mean bundle type?

Reply:
I meant file as in the type.json file. I can make that more explicit.